Loop/Managers/Alerts/StoredAlert.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  StoredAlert.swift
+//  Loop
+//
+//  Created by Rick Pasetto on 5/11/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import CoreData
+import LoopKit
+
+extension StoredAlert {
+
+    static var encoder = JSONEncoder()
+          
+    convenience init(from deviceAlert: DeviceAlert, context: NSManagedObjectContext, issuedTimestamp: Date = Date()) {
+        do {
+            self.init(context: context)
+            self.issuedTimestamp = issuedTimestamp
+            identifier = deviceAlert.identifier.value
+            // Encode as JSON strings
+            let encoder = StoredAlert.encoder
+            trigger = try encoder.encodeToStringIfPresent(deviceAlert.trigger)
```

Comment:
[nit] Why encode these to `string` and not binary `Data`? I suppose the same question for why using the `JSONEncoder`?

Reply:
For small amounts of data, I find it handy to encode in something human-readable, so I can double-check data by looking at the database, and it is easier to write unit test.  But this is totally up for debate; changing this later will be hard (because it would require a migration).  WDYT?

Reply:
I suppose it depends upon how much JSON would be generated for this. Either way, up to you.